e2e_tests/tests/experiment/test_custom_searcher.py

```diff
@@ -125,35 +126,71 @@ def test_run_random_searcher_exp() -> None:
 
 
 @pytest.mark.e2e_cpu_2a
-def test_resume_random_searcher_exp() -> None:
+@pytest.mark.parametrize(
+    "exceptions",
+    [
+        ["initial_operations_start", "progress_middle", "on_trial_closed_shutdown"],
+        ["on_validation_completed", "on_trial_closed_end", "on_trial_created_5"],
+        ["on_trial_created", "save_method_state", "after_save"],
+        [
+            "on_trial_created",
+            "save_method_state",
+            "load_method_state",
+            "after_save",
+            "after_save",
+            "on_validation_completed",
+            "after_save",
+            "save_method_state",
+        ],
+    ],
+)
+def test_resume_random_searcher_exp(exceptions: List[str]) -> None:
     config = conf.load_config(conf.fixtures_path("no_op/single.yaml"))
     config["searcher"] = {
         "name": "custom",
         "metric": "validation_error",
         "smaller_is_better": True,
         "unit": "batches",
     }
-    config["description"] = "custom searcher"
+    if len(exceptions) > 0:
+        config["description"] = ";".join(exceptions)
+    else:
+        config["description"] = "custom searcher"
 
     max_trials = 5
     max_concurrent_trials = 2
     max_length = 500
+    failures_expected = len(exceptions)
+    logging.info(f"expected_failures={failures_expected}")
 
     # do not use pytest tmp_path to experience LocalSearchRunner in the wild
     with tempfile.TemporaryDirectory() as searcher_dir:
         logging.info(f"searcher_dir type = {type(searcher_dir)}")
-        ex = MaxRetryError(
-            HTTPConnectionPool(host="dummyhost", port=8080),
-            "http://dummyurl",
-        )
-        search_method = RandomSearchMethod(max_trials, max_concurrent_trials, max_length, ex)
+        failures = 0
+        while failures < failures_expected:
+            try:
+                exception_step = exceptions.pop(0)
```

Comment:
If `exception_step` and `exception_point` mean the same thing maybe pick one name? I like `exception_point`.

Reply:
good catch. `exception_point` it is